mantidproject/mantid
# Add FindPeaksConvolve Algorithm to FitPropertyBrowser

### Description of work

#### Summary of work


This PR adds the new `FIndPeaksConvolve` algorithm to the FitPropertyBrowser.

It has been implemented in a way that should make it easy to add other peak fitting algorithms to the browser in the future.

I've also fixed all `cppcheck` warnings raised by the CI.

### To test:

1) Load the `EMU00020884 dataset from the training data (picked at random).
2) Plot spectrum 5, open the Fit Property Browser.
3) Open the `Setup` dropdown menu and select `FindPeaksConvolve`.
4) Enter the `EstimatedPeakExtent` as 0.1, then run the algorithm.
5) Observe found peaks.
6) Play around with other datasets, and different inputs to the algorithm.



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
